ships/0023-surface-results-in-buildruns.md

```diff
@@ -0,0 +1,304 @@
+| shp-source-default-commit-sha     |  `shp-source-${sourceName}-${resultName}`   | yes     |  at runtime | `${sourceName}` is `default` as `spec.source` has no name. |
+
+#### Future State of Build Results for Sources
+
+Assuming the following Build definition:
+
+```yaml
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: a-build
+spec:
+  sources:
+  - name: a-git-source
+    type: git
+    url: https://github.com/shipwright-io/sample-go
+  - name: an-http-source
+    type: http
+    url: https://licenses.company.com/customer-id/license.tar
+```
+
+We will have the following:
+
+| Result Name                       | Naming Convention                           | Emitted Today | Definition  |
+| --------------------------------: | ------------------------------------------: |-------: | ----------: |
+| shp-source-a-git-source-commit-sha     |  `shp-source-${spec.sources[0].name}-${resultName}` | no     |  at runtime |
+| shp-source-a-git-source-commit-author     |  `shp-source-${spec.sources[0].name}-${resultName}` | no     |  at runtime |
+| shp-source-an-http-source-size     |  `shp-source-${spec.sources[1].name}-${resultName}` | no     |  at runtime |
+
+#### Build Results Sources API Proposal
+
+Because of the future adoption of sources, the proposed API for `Build Results` should consider the following:
+
+- The _.status_ subresource should provide a relationship between a source and the results emitted in it´s step definition.
+- The _.status_ subresource should be able to surface results emitted from different sources.
+
+Therefore the **proposal** is to introduce `_.status.sources_` subresource which holds a list of `SourcesResult`:
+
+```go
+type BuildRunStatus struct { // This struct already exists on the BuildRun API, we just add a new field
+  Sources []SourceResult `json:"sources"`
+}
+
+type SourceResult struct {
+  Name   string             `json:"name"`
+  Git    *GitSourceResult   `json:"git,omitempty"`
+  Http   *HttpSourceResult  `json:"http,omitempty"` // A new field can be always introduced to support new types, e.g. bundle
+}
+
+type GitSourceResult struct{
+  CommitSha     string `json:"commit-sha,omitempty"`
+  CommitAuthor  string `json:"commit-author,omitempty"`
+}
+
+type HttpSourceResult struct{
+  Size int `json:"size"`
+}
+```
+
+A BuildRun will surface these results as follows:
+
+```yaml
+status:
+  sources:
+  - name: default
+    git:
+      commit-sha: cf156cfb0da9be144a40c7d5a110487e5c524
+```
+
+#### Build Results Output API Proposal
+
+Today, we only have a single output step in our strategies. There is no plan to modify this, therefore we propose here to introduce the _.status.output_ subresource in a BuildRun. This will require the following structures:
+
+```go
+type BuildRunStatus struct { // This struct already exists on the BuildRun API, we just add a new field
+  Output *Output `json:"output"`
+}
+
+type Output struct {
+  DigestOutputResult string `json:"digest,omitempty"`
```

Comment:
To conform with [our rule that the Go field should be called just like the JSON property](https://github.com/shipwright-io/build/pull/687).

```suggestion
  Digest string `json:"digest,omitempty"`
```

Reply:
Agree that we should adhere to this in the implementation, I don't think we should block merge of the proposal.